src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/GTKEngine.java

```diff
@@ -609,15 +602,27 @@ public BufferedImage finishPainting(boolean useCache) {
         WritableRaster raster = Raster.createPackedRaster(
                 dataBuffer, w0, h0, w0, bands, null);
 
-        ColorModel cm = COLOR_MODELS[transparency - 1];
-        BufferedImage img = new BufferedImage(cm, raster, false, null);
+        ColorModel cm = colorModelFor(transparency);
+        BufferedImage img = new BufferedImage(cm, raster, true, null);
         if (useCache) {
             cache.setImage(getClass(), null, w0, h0, cacheArgs, img);
         }
         graphics.drawImage(img, x0, y0, null);
         return img;
     }
 
+    private ColorModel colorModelFor(int transparency) {
+        synchronized (COLOR_MODELS) {
+            int index = transparency - 1;
+            if (COLOR_MODELS[index] == null) {
+                COLOR_MODELS[index] = GraphicsEnvironment.getLocalGraphicsEnvironment()
```

Comment:
How will it work if the configuration changes at runtime?

Reply:
You are right - as long as the old and new formats are mostly compatible the impact should be minimal. But in the worst case we might fall back to some slower blits. However with the current DirectColorModel, we are likely to hit optimized paths for most target formats.

Reply:
@mkartashev Did you check that the scenario described above cannot happen?